OpenRailAssociation/osrd
# Create slice STDCM and simulation#5272

Closes #5272

### Description
This marks the initial phase of decoupling the simulation functionality from `stdcm`. We are introducing two separate slices to handle each aspect independently. While we haven't completely replaced `osrdconf` at this stage, we've integrated it alongside our two new state slices. The end goal is to gradually deplete `osrdconf` and fully transition to the new slices.

### Key Highlights:
1. **Common Reducers Object** : We've created a shared reducers object that both slices will utilize, promoting code modularity and maintainability.
2. **updateSwitchTypes Action Refactoring** : Instead of triggering the action each time the infrastructure ID changes, which was leading to unnecessary calls, we've now streamlined it. We directly invoke the RTK query to fetch switch types precisely when and where they are needed.
3. osrdConf context :  We've implemented a react context to specify from which slice (stdcmconf or simulationconf), a component have to call a osrdconf actions 
4. **Testing** : We've added tests for both of the new slices.

### How to test ?

1. Start the application.
2. Inspect the Redux store to verify the presence of two new fields: `stdcmconf` and `simulationconf` in the state.
3. Navigate to the stdcm view and examine the `stdcmconf` state to confirm the existence of an `infraID`. In contrast, inspect the `simulationconf` state to verify the absence of infraID.


### Next step
Find a way to test the interactions between `updateInfraID`, `getInfraByIdSwitchTypes`. This involves mocking the RTK query for efficient testing. Following resources can help : 
  - https://dev.to/ifeanyichima/-testing-components-with-a-request-for-rtk-query-using-msw-and-react-testing-library-5a8n
  - https://remarkablemark.medium.com/how-to-test-redux-toolkit-query-with-jest-and-react-testing-library-cc02c0528fcf
  - https://gustavocd.dev/posts/testing-rtk-query-with-msw/

Additionally, we plan to replace occurrences of osrdConf.simulationConf with simulationConf and osrdConf.stdcmConf with stdcmConf in the store to maintain consistency and clarity (as noted in issues #5273 and #5274).
